src/Microsoft.Azure.WebJobs.Host/Bindings/BindingProviders/AsyncCollectorBindingProvider.cs

```diff
@@ -316,7 +310,8 @@ private class ExactBinding<TMessage> : BindingBase<TAttribute>
                     };
                 }
 
-                var cloner = new AttributeCloner<TAttribute>(attributeSource, context.BindingDataContract, parent._nameResolver, hookWrapper);
+                var clonerCtx = AttributeClonerContext.New(parent._nameResolver, context.Parameter);
```

Comment:
since this occurs in so many places, I think I'd prefer an overload of AttributeCloner which accepts the name resolver and parameter